base/iobuffer.jl

```diff
@@ -456,7 +456,7 @@ function take!(io::IOBuffer)
         if nbytes == 0 || io.reinit
             data = StringVector(0)
         elseif io.writable
-            data = wrap(Array, MemoryRef(io.data, io.offset + 1), nbytes)
+            data = view(io.data, io.offset+1:nbytes+io.offset+1)
```

Comment:
This seems to produce a `view` of length `nbytes+1`. Is that intended?

Reply:
```suggestion
            data = view(io.data, io.offset+1:nbytes+io.offset)
```

Reply:
Thank you. That was not intended.